auth/providers/azure/azure.go

```diff
@@ -139,8 +132,32 @@ func getMetadata(aadEndpoint, tenantID string) (*metadataJSON, error) {
 		Backoff:      retryablehttp.DefaultBackoff,
```

Comment:
where are we setting the max no of retries here as 3?

Reply:
It's 3 attempts, not 3 retries.

```go
RetryMax: 2, // initial + 2 retries = 3 attempts
```